pkg/events/derive/container_removed.go

```diff
@@ -0,0 +1,24 @@
+package derive
+
+import (
+	"github.com/aquasecurity/tracee/pkg/containers"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/events/parse"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// ContainerRemoved receives a containers.Containers object as a closure argument to track it's containers.
+// If it receives a cgroup_rmdir event, it can derive a container_remove event from it.
+func ContainerRemoved(containers *containers.Containers) events.DeriveFunction {
+	argsGen := func(event trace.Event) ([]interface{}, error) {
+		cgroupId, err := parse.ArgUint64Val(&event, "cgroup_id")
+		if err != nil {
+			return nil, err
+		}
+		if info := containers.GetCgroupInfo(cgroupId); info.Container.ContainerId != "" {
+			return []interface{}{cgroupId}, nil
+		}
+		return []interface{}{}, nil
+	}
+	return generateSingleEventDeriveFunction(events.ContainerRemove, argsGen)
```

Comment:
Just a thought, you can do an an anonymous function here instead so it would read:

```
return generateSingleEventDeriveFunction(events.ContainerRemove, func(event trace.Event) ([]interface{}, error) {
  cgroupId, err := parse.ArgUint64Val(&event, "cgroup_id")
  if err != nil {
	  return nil, err
  }
  if info := containers.GetCgroupInfo(cgroupId); info.Container.ContainerId != "" {
	  return []interface{}{cgroupId}, nil
  }
  return []interface{}{}, nil
})
```

This pattern is very common in javascript actually, and usually that's the syntax used.
Up to you what you think looks better though.

Reply:
So after thinking about it more I decided that the function here should be testable.
So I decided that it would be better to create her in a sub-function.